pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -289,6 +293,158 @@ func TestReconcile(t *testing.T) {
+		},
+		{
+			name: "cluster not yet due for hibernate no condition", // cluster that has never been hibernated and thus has no running condition
+			cd: cdBuilder.GenericOptions(testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "12h")).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectRequeueAfter: true,
+			expectedPowerState: "",
+		},
+		{
+			name: "cluster with running condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.RunningHibernationReason, 9*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster with running condition not due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "20h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.RunningHibernationReason, 6*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectRequeueAfter: true,
+			expectedPowerState: "",
+		},
+		{
+			name: "cluster stuck resuming condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				// TODO: check false below
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.ResumingHibernationReason, 9*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster failed resuming condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				// TODO:
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.FailedToStartHibernationReason, 9*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster waking from hibernate",
+			setupActuator: func(actuator *mock.MockHibernationActuator) {
+				actuator.EXPECT().MachinesRunning(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(false, nil)
+			},
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "4h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionTrue, hivev1.ResumingHibernationReason, 8*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10*time.Hour)),
+				o.shouldRun).Build(),
+			expectedPowerState: hivev1.RunningClusterPowerState,
+			expectRequeueAfter: true,
```

Comment:
This test case highlights a deficiency in the tests. Since the test only checks whether the requeue-after is positive, the test does not distinguish between a requeue-after coming from the hibernate-after annotation or a requeue-after for other reasons. This test case should be verifying that the requeue-after did not get set as a result of the hibernate-after annotation, but it does not verify that. The requeue-after is getting set to `stateCheckInterval`.

Reply:
Moved this to fuzzy matching, asserting it's within 10s of expected.